src/Grid/grid.jl

```diff
@@ -2,18 +2,30 @@
 # Main types for meshes #
 #########################
 """
+    Node{dim, T}
+
 A `Node` is a point in space.
+
+# Fields
+- `x::Vec{dim,T}`: stores the coordinates
 """
 struct Node{dim,T}
     x::Vec{dim,T}
 end
 Node(x::NTuple{dim,T}) where {dim,T} = Node(Vec{dim,T}(x))
 getcoordinates(n::Node) = n.x
 
+
+abstract type AbstractCell{dim,N,M} end
 """
+    Cell{dim,N,M} <: AbstractCell{dim,N,M}
 A `Cell` is a sub-domain defined by a collection of `Node`s as it's vertices.
+However, a `cell` is not defined by the nodes but rather by the global node ids.
+The parameter `dim` refers here to the geometrical/ambient dimension, i.e. the dimension of the `nodes` in the grid and **not** the topological dimension of the cell.
+
```

Comment:
Perhaps add a comment about what `N` and `M` refer to?
```suggestion
A `Cell` has `N` nodes and `M` faces.

```